### ISSUE TITLE
tf.divide handles "name" argument differently

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#6741]
If installed from binary pip package, provide:

1. A link to the pip package you installed:
https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-linux-gpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=gpu-linux/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow_gpu-0.12.1-cp27-none-linux_x86_64.whl
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
0.12.head

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
```python
a = tf.get_variable('a', shape=[10])
b = tf.get_variable('b', shape=[10])
print tf.divide(a, b, name='x').name   # x/truediv:0
```

Normally one would expect (like almost every other existing function in tf) that with the name='x' option, the output op will be named 'x:0'. 
But instead it was named "x/truediv:0".

**Comment by yaroslavvb** [comment]
So what's happening here is that `divide` tries to do version-sensitive Python2 or Python 3 style divide. It calls `a/b` which drops the name, and then gets redirected to Python2 or Python3  . I think you probably want `tf.div` instead of `tf.divide`.

**Comment by yaroslavvb** [comment]
@aselle for comment if this is likely to get fixed .... it seems `tf.divide` loses the name which makes annoying for graph/timeline visualization

**Comment by aselle** [comment]
I think I have a fix. The problem is that operator overload __div__ cannot pass name, but I do want to use a class' division dispatch to honor the Python2/Python3 modes. I can do that with a dummy class. I have a fix that I'm testing internally. WIll keep this bug updated.

**Comment by aselle** [comment]
@ppwwyyxx, master now has the fix, You can cherry pick it if you need it in an older version, otherwise in the mean-time you can use the workaround. Thanks!

**Comment by ppwwyyxx** [comment]
Great! Thanks a lot!

**Comment by ppwwyyxx** [comment]
Similar problem was found in `tf.case` and `tf.train.piecewise_constant`.

**Comment by yaroslavvb** [comment]
I think they are likely to get fixed if they are filed as separate issues (so they get triaged properly). Good to reference this issue so people can refer back to the commit that fixes things

**Comment by ppwwyyxx** [comment]
👍  I was hesitated to open too many similar issues..
If that's fine I'll open one for each
